data/preprocessing/nutrient_load_reduction/process_data.py

```diff
@@ -0,0 +1,92 @@
+""" Reads the limiting nutrients equal area vector file, reporjects the file to EPSG4326 and estimates the percentage of reduction needed to meet a good water quality conditions.
+
+Usage:
+process_data.py <folder>
+
+Arguments:
+    <folder>     Folder containing the limiting nutrients shapefile
+"""
+import os
+import logging
+from pathlib import Path
+import argparse
+
+import geopandas as gpd
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("preprocessing_limiting_nutrients_file")
+
+def check_and_reproject_to_4326(gdf):
+    """
+    Checks if a GeoDataFrame is in CRS 4326 (WGS84) and reprojects it if not.
+
+    Parameters:
+    - gdf: GeoDataFrame to check and reproject if needed.
+
+    Returns:
+    - Reprojected GeoDataFrame (if reprojected) or the original GeoDataFrame (if already in 4326).
+    """
+    if gdf.crs is None or gdf.crs.to_epsg() != 4326:
+        log.info("Reprojecting GeoDataFrame to EPSG:4326 (WGS84)...")
+        try:
+            # Reproject to EPSG:4326
+            gdf = gdf.to_crs(epsg=4326)
+            log.info("Reprojection successful.")
+        except:
+            log.error("Reprojection failed with error")
+    else:
+        log.info("GeoDataFrame is already in EPSG:4326 (WGS84).")
```

Comment:
a bit nit pick but it can be either 4326 or not have a crs

Reply:
I'm filtering the crs is None above.